lib/cloud_controller/deployment_updater/actions/cancel.rb

```diff
@@ -0,0 +1,41 @@
+require 'cloud_controller/deployment_updater/calculators/find_interim_web_process'
+require 'cloud_controller/deployment_updater/actions/cleanup_web_processes'
+
+module VCAP::CloudController
+  module DeploymentUpdater
+    module Actions
+      class Cancel
+        attr_reader :deployment, :logger, :app, :deploying_web_process
+
+        def initialize(deployment, logger)
+          @deployment = deployment
+          @logger = logger
+          @app = deployment.app
+          @deploying_web_process = deployment.deploying_web_process
+        end
+
+        def call
+          deployment.db.transaction do
+            app.lock!
+            return unless deployment.lock!.state == DeploymentModel::CANCELING_STATE
+
+            deploying_web_process.lock!
+
+            prior_web_process = Calculators::FindInterimWebProcess.new(deployment).call || app.oldest_web_process
+            prior_web_process.lock!
+
+            prior_web_process.update(instances: deployment.original_web_process_instance_count, type: ProcessTypes::WEB)
```

Comment:
**question** Again existing behaviour, but does this mean if I do a deployment, then supersede it with another deployment, and then cancel my new deployment, it will revert to the superseded deployment with the desired number of instances?

Reply:
I believe so.  I'll have to try it to be sure.  I'm beginning to think we need a doc to track the actual behavior of different Deployment scenarios

Reply:
I wonder if thats expected, I think you could argue it should instead restore to the last known complete deployment/ original web processes.